Fluid error / crash with GT 1.6.1 and Pneumaticcraft Repressurized

### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

1.6.1

### Minecraft Version

1.20.1 Forge

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

Yes

### Other Installed Mods

- JEI v15.20.0.105
- Pneumaticcraft v6.0.18
- GTCEu-Modern v1.6.1
- Patchouli v1.20.1-84

### Expected Behavior

Expected the update from 1.5.4 to 1.6.1 to not change Pneumaticcraft behavior so that it fails to register the Block Heat Properties recipe category in JEI.

### Actual Behavior

Updated to 1.6.1 and Pneumaticcraft's Block Heat Properties fail to register now.

### Steps to Reproduce

1) Install JEI, Patchouli, Pneumaticcraft, and GTCEu-Modern
2) Look at the uses for Heat Pipe
3) Observe the lack of a Block Heat Properties tab in JEI

### Additional Information

The Block Heat Properties tab populates correctly when using GTCEu-Modern 1.5.4, and disappears after updating which is why I posted this issue here - let me know if you think it should go to PNC! I haven't been able to recreate the associated crash yet. I suspect if you are on 1.5.4 and try to set up something utilizing the Block Heat Properties from PNC and then update to 1.6.1 that the crash will happen. Either way, the error appears to be similar between a crash and just JEI registering the recipes.
```[13:31:39] [Render thread/ERROR]: Caught an error from mod plugin: class me.desht.pneumaticcraft.common.thirdparty.jei.JEIPlugin pneumaticcraft:default
java.lang.IllegalArgumentException: Cannot set property IntegerProperty{name=level, clazz=class java.lang.Integer, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]} to 16 on Block{minecraft:water}, it is not an allowed value
	at net.minecraft.world.level.block.state.StateHolder.m_61124_(StateHolder.java:130) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraftforge.fluids.ForgeFlowingFluid.m_5804_(ForgeFlowingFluid.java:138) ~[forge-1.20.1-47.3.11-universal.jar%23169!/:?]
	at net.minecraft.world.level.material.FluidState.m_76188_(FluidState.java:96) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.registerDefaultFluidValues(BlockHeatProperties.java:90) ~[pneumaticcraft-repressurized-6.0.18+mc1.20.1.jar%23163!/:6.0.18+mc1.20.1]
	at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.populateCustomHeatEntries(BlockHeatProperties.java:78) ~[pneumaticcraft-repressurized-6.0.18+mc1.20.1.jar%23163!/:6.0.18+mc1.20.1]
	at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.getAllEntries(BlockHeatProperties.java:61) ~[pneumaticcraft-repressurized-6.0.18+mc1.20.1.jar%23163!/:6.0.18+mc1.20.1]
	at me.desht.pneumaticcraft.common.thirdparty.jei.JEIBlockHeatPropertiesCategory.getAllRecipes(JEIBlockHeatPropertiesCategory.java:88) ~[pneumaticcraft-repressurized-6.0.18+mc1.20.1.jar%23163!/:6.0.18+mc1.20.1]
	at me.desht.pneumaticcraft.common.thirdparty.jei.JEIPlugin.registerRecipes(JEIPlugin.java:141) ~[pneumaticcraft-repressurized-6.0.18+mc1.20.1.jar%23163!/:6.0.18+mc1.20.1]
	at mezz.jei.library.load.PluginLoader.lambda$createRecipeManager$13(PluginLoader.java:182) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at mezz.jei.library.load.PluginCaller.callOnPlugins(PluginCaller.java:25) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at mezz.jei.library.load.PluginLoader.createRecipeManager(PluginLoader.java:182) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at mezz.jei.library.startup.JeiStarter.start(JeiStarter.java:115) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at mezz.jei.forge.startup.StartEventObserver.transitionState(StartEventObserver.java:130) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at mezz.jei.forge.startup.StartEventObserver.onEvent(StartEventObserver.java:98) ~[jei-1.20.1-forge-15.20.0.105.jar%23161!/:15.20.0.105]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:260) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:252) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_5859_(ClientPacketListener.java:1451) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateTagsPacket.m_5797_(ClientboundUpdateTagsPacket.java:35) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateTagsPacket.m_5797_(ClientboundUpdateTagsPacket.java:11) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_263899_(PacketUtils.java:22) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:115) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1106) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:718) ~[client-1.20.1-20230612.114412-srg.jar%23164!/:?]
	at net.minecraft.client.main.Main.main(Main.java:218) ~[forge-47.3.11.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111) ~[fmlloader-1.20.1-47.3.11.jar:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:99) ~[fmlloader-1.20.1-47.3.11.jar:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$makeService$0(CommonClientLaunchHandler.java:25) ~[fmlloader-1.20.1-47.3.11.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
```

Compared to a crash:
```[12:59:13] [Server thread/ERROR]: Encountered an unexpected exception
--
6069 | net.minecraft.ReportedException: Exception ticking world
6070 | at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:897) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6071 | at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6072 | at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:89) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6073 | at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6074 | at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6075 | at java.lang.Thread.run(Thread.java:833) ~[?:?]
6076 | Caused by: java.lang.IllegalArgumentException: Cannot set property IntegerProperty{name=level, clazz=class java.lang.Integer, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]} to 16 on Block{minecraft:water}, it is not an allowed value
6077 | at net.minecraft.world.level.block.state.StateHolder.m_61124_(StateHolder.java:130) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6078 | at net.minecraftforge.fluids.ForgeFlowingFluid.m_5804_(ForgeFlowingFluid.java:138) ~[forge-1.20.1-47.3.11-universal.jar%231013!/:?]
6079 | at net.minecraft.world.level.material.FluidState.m_76188_(FluidState.java:96) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6080 | at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.registerDefaultFluidValues(BlockHeatProperties.java:90) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6081 | at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.populateCustomHeatEntries(BlockHeatProperties.java:78) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6082 | at me.desht.pneumaticcraft.common.heat.BlockHeatProperties.getCustomHeatEntry(BlockHeatProperties.java:51) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6083 | at me.desht.pneumaticcraft.common.heat.behaviour.HeatBehaviourCustomTransition.initialize(HeatBehaviourCustomTransition.java:49) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6084 | at me.desht.pneumaticcraft.common.heat.behaviour.HeatBehaviourManager.addHeatBehaviours(HeatBehaviourManager.java:75) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6085 | at me.desht.pneumaticcraft.common.heat.HeatExchangerLogicTicking.initializeAsHull(HeatExchangerLogicTicking.java:67) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6086 | at me.desht.pneumaticcraft.common.block.entity.IHeatExchangingTE.lambda$initializeHullHeatExchangers$1(IHeatExchangingTE.java:91) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6087 | at java.util.IdentityHashMap.forEach(IdentityHashMap.java:1354) ~[?:?]
6088 | at me.desht.pneumaticcraft.common.block.entity.IHeatExchangingTE.initializeHullHeatExchangers(IHeatExchangingTE.java:90) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6089 | at me.desht.pneumaticcraft.common.block.entity.AbstractPneumaticCraftBlockEntity.onLoad(AbstractPneumaticCraftBlockEntity.java:244) ~[pneumaticcraft-repressurized-6.0.17+mc1.20.1.jar%23886!/:6.0.17+mc1.20.1]
6090 | at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
6091 | at net.minecraft.world.level.Level.m_46463_(Level.java:453) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6092 | at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:351) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6093 | at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:893) ~[client-1.20.1-20230612.114412-srg.jar%231008!/:?]
6094 | ... 5 more
6095 | [12:59:13] [Server thread/FATAL]: Preparing crash report with UUID c7fa493a-5767-47a7-93d5-d86d72edeb21```

--- a/src/main/java/com/gregtechceu/gtceu/common/fluid/potion/PotionFluid.java
+++ b/src/main/java/com/gregtechceu/gtceu/common/fluid/potion/PotionFluid.java
@@ -14,6 +14,8 @@ import net.minecraft.world.item.alchemy.PotionUtils;
 import net.minecraft.world.item.alchemy.Potions;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.LiquidBlock;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.FluidState;
 import net.minecraftforge.client.extensions.common.IClientFluidTypeExtensions;
 import net.minecraftforge.fluids.FluidStack;
@@ -30,6 +32,13 @@ public class PotionFluid extends ForgeFlowingFluid {
         super(properties
                 .bucket(() -> Items.AIR)
                 .block(() -> (LiquidBlock) Blocks.WATER));
+        registerDefaultState(getStateDefinition().any().setValue(LEVEL, 7));
+    }
+
+    @Override
+    protected void createFluidStateDefinition(StateDefinition.Builder<Fluid, FluidState> builder) {
+        super.createFluidStateDefinition(builder);
+        builder.add(LEVEL);
     }
 
     public static FluidStack of(int amount, Potion potion) {
@@ -74,7 +83,7 @@ public class PotionFluid extends ForgeFlowingFluid {
 
     @Override
     public int getAmount(FluidState state) {
-        return 0;
+        return state.getValue(LEVEL);
     }
 
     public static class PotionFluidType extends FluidType {

